src/plugins/layout.ts

```diff
@@ -117,13 +147,35 @@ export default class Layout implements IPlugin {
     }
   }
 
+  private cssQueue(element: Element) {
+    // Clear the timeout if it already exists
+    if (this.cssTimeout) {
+      clearTimeout(this.cssTimeout);
+    }
+
+    // Queue element to be processed after the timeout triggers
+    if (this.cssElementQueue.indexOf(element) === -1) {
+      this.cssElementQueue.push(element);
+    }
+
+    this.cssTimeout = setTimeout(this.cssDequeue.bind(this), this.cssTimeoutLength);
+  }
+
+  private cssDequeue() {
+    for (let element of this.cssElementQueue) {
+      this.layoutHandler.bind(this, element, Source.Css);
+    }
+  }
+
   private layoutHandler(element: Element, source: Source) {
     let index = getNodeIndex(element);
     let nodeInfo = this.shadowDom.getNodeInfo(index);
     if (nodeInfo) {
       let layoutState = nodeInfo.state as IElementLayoutState;
+      let styleState = nodeInfo.state as IStyleLayoutState;
       switch (source) {
         case Source.Scroll:
+          layoutState = layoutState as IElementLayoutState;
```

Comment:
>layoutState = layoutState as IElementLayoutState; [](start = 10, length = 49)

Isn't layoutstate already typed as IElementLayoutState at line #174 ?

Reply:
Fixed

---
In reply to: [243076492](https://github.com/Microsoft/clarity-js/pull/116#discussion_r243076492) [](ancestors = 243076492)